elyra/pipeline/parser.py

```diff
@@ -58,32 +58,30 @@ def parse(pipeline_definition) -> Pipeline:
             raise ValueError("Invalid pipeline: At least one node must exist in primary pipeline.")
 
         pipeline_object = Pipeline(pipeline['id'],
-                                   PipelineParser._read_pipeline_title(pipeline),
+                                   PipelineParser._read_pipeline_name(pipeline),
                                    PipelineParser._read_pipeline_runtime(pipeline),
-                                   PipelineParser._read_pipeline_runtime_config(pipeline),
-                                   PipelineParser._read_pipeline_filetype(pipeline),
-                                   PipelineParser._read_pipeline_export(pipeline))
+                                   PipelineParser._read_pipeline_runtime_config(pipeline))
 
         for node in pipeline['nodes']:
             # Supernodes are not supported
             if node['type'] == "super_node":
                 raise ValueError('Invalid pipeline: Supernode feature is not supported.')
 
             # parse links as dependencies
-            links = PipelineParser._read_pipeline_operation_dependencies(node)
+            links = PipelineParser._read_pipeline_parent_operation_dependencies(node)
 
             # parse each node as a pipeline operation
             operation = Operation(
                 id=node['id'],
                 type=node['type'],
-                title=node['app_data']['ui_data']['label'],
-                artifact=node['app_data']['artifact'],
-                image=node['app_data']['image'],
-                vars=node['app_data'].get('vars') or [],
-                file_dependencies=node['app_data'].get('dependencies') or [],
-                recursive_dependencies=node['app_data'].get('recursive_dependencies') or False,
+                classification=node['op'],
+                filename=node['app_data']['filename'],
+                runtime_image=node['app_data']['runtime_image'],
+                dependencies=node['app_data'].get('dependencies') or [],
```

Comment:
This error is currently propagating to the user, but `KeyError` is a more confusing error to the user than `ValueError` (at least when reading their stack trace)

Reply:
Ok, I am adding `or None` to all remaining fields

Reply:
Unfortunately, that won't work.  You'll need to switch to `node.get('op')` so that it even gets into the constructor.  Just adding `or None` won't work because the first portion raises `KeyError` before the or-clause is processed.
```
Python 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 14:38:56) 

>>> foo = {"bar" : {}}
>>> res = foo['bar']['name'] or None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'name'
>>> res = foo['bar'].get('name')
>>> res is None
True
```

Reply:
Ok, thank you.